src/pages/workspaces/workspace/analysis/file-utils.ts

```diff
@@ -0,0 +1,113 @@
+import _ from 'lodash/fp'
+import { useState } from 'react'
+import { Ajax } from 'src/libs/ajax'
+import { withErrorReporting } from 'src/libs/error'
+import { useCancellation, useOnMount, useStore } from 'src/libs/react-utils'
+import { workspaceStore } from 'src/libs/state'
+import * as Utils from 'src/libs/utils'
+import {
+  runtimeTools,
+  ToolLabel,
+  toolToExtensionMap
+} from 'src/pages/workspaces/workspace/analysis/tool-utils'
+import {
+  CloudProviderType,
+  isGoogleWorkspaceInfo,
+  WorkspaceWrapper
+} from 'src/pages/workspaces/workspace/workspace-utils'
+
+// This syntax gives us nominal typing
+// Nominal types are not interchangable even if the underlying type is the same
+// A func that takes (a: A, b: B) where A and B are both strings cannot be called like (b,a) with this restriction
+export type NominalType<BaseType, Name extends string> = BaseType & { __typeToken: Name }
+
+export type FileName = NominalType<string, 'FileName'> // represents a file with an extension and no path, eg `dir/file.ipynb` =>  `file.ipynb`
+export type AbsolutePath = NominalType<string, 'AbsolutePath'> // represents an absolute path in the context of a cloud storage directory structure, i.e. `dir/file.ipynb`
+export type Extension = NominalType<string, 'Extension'> // represents the substring found after the last dot in a file, eg `dir/file.txt.ipynb` => `ipynb`
+export type DisplayName = NominalType<string, 'DisplayName'> // represents the name of a file without an extension or pathing eg `dir/file.txt.ipynb`=> `dir/file.txt`
+
+// removes all paths up to and including the last slash
+export const getFileName = (path: string): FileName => _.flow(_.split('/'), _.last)(path) as FileName
+
+export const getExtension = (path: string): Extension => _.flow(_.split('.'), _.last)(path) as Extension
+
+// ex abs/file.ipynb -> abs/file
+export const stripExtension = (path: string): string => _.replace(/\.[^/.]+$/, '', path)
+
+// removes leading dirs and a file ext suffix on paths
+export const getDisplayName = (path: string): DisplayName => _.flow(getFileName, stripExtension)(path) as DisplayName
+
+export interface FileMetadata {
+  lockExpiresAt: string
+  lastLockedBy: string
+  hashedOwnerEmail?: string
+}
+
+export interface AnalysisFile {
+  name: AbsolutePath
+  ext: Extension
+  displayName: DisplayName
+  fileName: FileName
+  lastModified: number
+  tool: ToolLabel
+  cloudProvider: CloudProviderType
+  metadata?: FileMetadata
+}
+
+export const analysisFilesStore = Utils.atom()
+
+export interface AnalysisFileStore {
+  analyses: AnalysisFile[],
+  refresh: unknown //TODO: How can we better type this
+  loading: boolean
+}
+
+export const useAnalysisFiles = (): AnalysisFileStore => {
+  const signal = useCancellation()
+  const [loading, setLoading] = useState(false)
+  const workspace: WorkspaceWrapper = useStore(workspaceStore)
+  const analyses: AnalysisFile[] = useStore(analysisFilesStore)
```

Comment:
I followed the pattern that `useWorkspaces` in `workspace-utils` does. Since `workspaceStore` is global, I think it makes sense for `analysisFilesStore` to be global since `analysisFiles` are used in multiple (3) components and are directly derived from the workspace in `workspaceStore`. Worth noting I kept this code in this PR since it is a file util, but it isn't actually used (or tested) until the subsequent PR (see current AnalysisDuplicator logic which calls the APIs that this component does directly). Could be convinced to move it out for now...

Reply:
Generally, state adds complexity. Since this hook refreshes on mount anyway, the only value that I see the global store providing is caching. However, because the store is global but the state is per-workspace, now we need to deal with resetting this when the workspace changes. As is, it looks like if a user looked at this in workspace A, then workspace B, they'd see analyses for workspace A in workspace B until the refresh request returned.

Reply:
Also, since this is more closely tied to refactoring the AnalysisDuplicator in a subsequent PR than the changes to notebook-utils that make up the bulk of this PR, I would have suggested putting in that PR instead of this one.

Reply:
Yeah, I was starting to think the same. I'll remove it for now.

Reply:
It should refresh on mount in any component that uses it... probably needs more testing in the subsequent PR though since I'm not super familiar with this `store` pattern and this is mostly copy/paste